feat(tasks/coverage): add acorn-jsx conformance

hi-ogawa opened the pull request:
Part of https://github.com/oxc-project/oxc/issues/9703
Requires https://github.com/oxc-project/acorn-test262/pull/3

hi-ogawa commented:
* **#9745** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9745?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9745?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9745).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
@hi-ogawa Is there a reason this is draft? All looks good to me.

https://github.com/oxc-project/acorn-test262/pull/3 has been merged.

hi-ogawa commented:
Maybe I thought I should make use of `test-acorn-jsx/fail` tests to check expected failures, but if you think that's not necessary or can be done later, please feel free to merge it :+1:

hi-ogawa committed:
feat(tasks/coverage): add acorn-jsx conformance (#9745)

Part of https://github.com/oxc-project/oxc/issues/9703
Requires https://github.com/oxc-project/acorn-test262/pull/3
